tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/ExpressionTypeTestNew.java

```diff
@@ -135,6 +136,23 @@ public void testArrayLiteral() {
         assertEquals(memberType.typeKind(), STRING);
     }
 
+    @Test(dataProvider = "RegexpTemplateLiteralPosProvider")
+    public void testRegexpTemplateLiteralExpr(int sLine, int sCol, int eLine, int eCol) {
+        TypeSymbol type = getExprType(sLine, sCol, eLine, eCol);
+        assertEquals(type.typeKind(), TYPE_REFERENCE);
```

Comment:
Should we validate signature as well?

Reply:
Awaiting changes: https://github.com/ballerina-platform/ballerina-lang/pull/39509/commits/973652b34a52057d12e7ff7d57cf32da8b7fb56f from #39509.